.github/workflows/ci-go.yaml

```diff
@@ -0,0 +1,29 @@
+name: Go Continuous Integration
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+jobs:
+  golangci-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '1.21'
```

Comment:
You can use the go version from the mod file, instead of 'go-version', it's a different flag 'go-version-file'